### PR TITLE
Issue 500 feature add reference secret manager implementation demoapps

### PR DESCRIPTION
**Pull request by marianod92**
## What?
Add example implementation of Secrets Manager and how to consume secrets from EKS

## Why?
To have basic implementation as an example for using Secrets Manager from some workload in EKS, particularly on the reference code of [k8s-eks-demoapps ](https://github.com/binbashar/le-tf-infra-aws/tree/master/apps-devstg/us-east-1/k8s-eks-demoapps)reference code.

### CODE REVIEW

File: apps-devstg/us-east-1/k8s-eks-demoapps/k8s-components/security.tf [hunk 1]
```diff
@@ -52,40 +52,42 @@ resource "helm_release" "external_secrets" {
...
+                  serviceAccountRef:
+                    name: external-secrets
+                    namespace: external-secrets
+    EOF
+  ]
+
+  depends_on = [helm_release.external_secrets[0]]
+}
+
 # These resources below (cluster_secrets_manager and cluster_parameter_store) need to be commented out and applied in a second step
```
Comment (Franr) [inline apps-devstg/us-east-1/k8s-eks-demoapps/k8s-components/security.tf:86]: I guess this comment doesn't apply anymore?

File: apps-devstg/us-east-1/k8s-eks-demoapps/identities/config.tf [hunk 2]
```diff
@@ -59,3 +70,14 @@ data "terraform_remote_state" "shared-dns" {
...
+data "terraform_remote_state" "security-keys" {
+  backend = "s3"
+
+  config = {
+    region  = var.region
+    profile = var.profile
+    bucket  = var.bucket
+    key     = "${var.environment}/security-keys/terraform.tfstate"
+  }
+}
```
Comment (Franr) [inline apps-devstg/us-east-1/k8s-eks-demoapps/identities/config.tf:83]:
```suggestion
}

```

File: apps-devstg/us-east-1/k8s-eks-demoapps/secrets/config.tf [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  backend = "s3"
+
+  config = {
+    region  = var.region
+    profile = var.profile
+    bucket  = var.bucket
+    key     = "${var.environment}/security-keys/terraform.tfstate"
+  }
+}
+# Note: Commented out while database is not deployed
```
Comment (Franr) [inline apps-devstg/us-east-1/k8s-eks-demoapps/secrets/config.tf:37]: unnecessary?

File: apps-devstg/us-east-1/k8s-eks-demoapps/identities/config.tf [hunk 4]
```diff
@@ -59,3 +70,14 @@ data "terraform_remote_state" "shared-dns" {
...
 }
+
+data "terraform_remote_state" "security-keys" {
+  backend = "s3"
+
+  config = {
+    region  = var.region
+    profile = var.profile
+    bucket  = var.bucket
+    key     = "${var.environment}/security-keys/terraform.tfstate"
```
Comment (Franr) [inline apps-devstg/us-east-1/k8s-eks-demoapps/identities/config.tf:81]: isn't this the same state than `apps-devsgt-keys`? shouldn't it be `security/.....` instead?

**Review by Franr — COMMENTED** [review]
left a few comments!

**Review by diego-ojeda-binbash — CHANGES_REQUESTED** [review]

**Review by Franr — APPROVED** [review]

**Comment by marianod92** [comment]
- Test after modifications

image
image
image

**Review by Franr — APPROVED** [review]

**Review by diego-ojeda-binbash — APPROVED** [review]

**Pull request merged**